### DISCUSSION TITLE
Pico serial port disappears after flashing

### DISCUSSION THREAD

**Discussion by w9ran** [earlephilhower/arduino-pico#2327] [General]
I'm working to move a multicore project that was originally compiled using the sdk over to the Arduino IDE using the Arduino-Pico core.   When I drop the original uf2 onto the Pico,  a diagnostic printf statement will produce output on the serial monitor via stdio.   But while my Arduino version compiles, downloads, and runs - after the flash is completed the serial port used for downloading has disappeared.  Unplugging-replugging doens't bring it back nor does the board show up in Device Manager.    Dropping the original uf2 restores normal operation.   The host is a PC running Win 11.

I've tried compiling with and without Serial.begin enabled, and have commented out all the printf statements but the serial port still vanishes no matter what.    

I thought the problem was related to conflict over USB serial but that seems to not be the case.   I did have to comment out this line to get the sdk-written code to compile:   // stdio_init_all();   I haven't had this happen with other sdk-developed code that I've moved to Arduino core, and the serial port also works as it should using Arduino serialprint statements.       Thanks for any suggestions!

**Comment by earlephilhower** [comment]
USB is run by code on the Pico core, so what is happening is your code is most likely crashing.  If the Pico crashes, it can't respond to the host PC's USB packets.

You can use a PicoDebug (aka Picoprobe) to hook up GDB to find exactly where it's crashed, or winnow it down to an MCVE manually.

**Reply by earlephilhower** [reply]
Also, don't use `stdio_init_all` (I think you already removed that). `Serial` owns the USB CDC port.

 `printf` (standard POSIX, not `Serial.printf`) only outputs in debug mode so you'll want to set Tools->Debug Port->Serial to redirect `printf(xxx)` output or it will just go to the bit bucket.

**Reply by w9ran** [reply]
Gotit, thanks!  I since confirmed that the core0 application is running normally but the core1 code is not, so that's the starting point.
